specification/app/resource-manager/Microsoft.App/preview/2023-08-01-preview/Patches.json

```diff
@@ -0,0 +1,555 @@
+        },
+        "newLayer": {
+          "type": "object",
+          "properties": {
+            "name": {
+              "type": "string",
+              "description": "The details of the new layer for the target image."
+            },
+            "frameworkAndVersion": {
+              "type": "string",
+              "description": "The framework and its version in the new run image for the target image."
+            },
+            "osAndVersion": {
+              "type": "string",
+              "description": "The OS name and its version in the new run image for the target image."
+            }
+          },
+          "description": "New layer update details in the target image.",
+          "readOnly": true
+        },
+        "oldLayer": {
+          "type": "object",
+          "properties": {
+            "name": {
+              "type": "string",
+              "description": "The details of the old layer for the target image."
+            },
+            "frameworkAndVersion": {
+              "type": "string",
+              "description": "The framework and its version in the old run image for the target image."
+            },
+            "osAndVersion": {
+              "type": "string",
+              "description": "The OS name and its version in the old run image for the target image."
+            }
+          },
+          "description": "The old layer details in the target image.",
+          "readOnly": true
+        },
+        "patchType": {
+          "type": "string",
+          "description": "The type for the patch.",
+          "enum": [
+            "FrameworkSecurity",
+            "OSSecurity",
+            "FrameworkAndOSSecurity",
+            "Other"
+          ],
+          "x-ms-enum": {
+            "name": "PatchType",
+            "modelAsString": true
+          },
+          "readOnly": true
+        }
+      },
+      "readOnly": true
+    },
+    "PatchProperties": {
+      "type": "object",
+      "properties": {
+        "targetEnvironmentId": {
+          "format": "arm-id",
+          "type": "string",
+          "description": "The id of the target environment for the patch."
+        },
+        "targetContainerAppId": {
+          "format": "arm-id",
+          "type": "string",
+          "description": "The id of the target container app for the patch."
+        },
+        "targetRevisionId": {
+          "format": "arm-id",
+          "type": "string",
+          "description": "The id of the target revision for the patch."
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/PatchProvisioningState",
+          "description": "Patch provisioning state.",
+          "readOnly": true
+        },
+        "patchApplyStatus": {
+          "$ref": "#/definitions/PatchApplyStatus",
+          "description": "The status of the patch operation once it has been provisioned",
+          "readOnly": true
+        },
+        "createdTime": {
```

Comment:
There’s already createTime and UpdateTime inside commontype/proxyResource/systemdata
And Patch resource contains proxyResource
 "description": "Container App Patch",
      "allOf": [
        {
          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
        }
      ]


It‘s dup createTime and updateTime here in my opinion

Reply:
Ref to https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json

Reply:
the main reason we have a createdTime here is because the systemData created time is not used in patches. The resource created time we have shows the resource creation time in data plane. Updated time shows the last time the resource got updated either by user or data plane. 

I believe this is also what container app revision was doing.